### PR TITLE
[FrameworkBundle] [Workflow] fixed initial place config

### PR DESCRIPTION
**Pull request by HeahDude**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/symfony/pull/20458#issuecomment-259950141
| License       | MIT
| Doc PR        | ~

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd [hunk 1]
```diff
@@ -238,6 +238,7 @@
         <xsd:sequence>
             <xsd:element name="marking-store" type="marking_store" />
             <xsd:element name="supports" type="xsd:string" minOccurs="1" maxOccurs="unbounded" />
+            <xsd:element name="initial_place" type="xsd:string" />
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd]: should be `<xsd:element name="initial_place" type="xsd:string" minOccurs="0" maxOccurs="1" />`

**Review by xabbuh — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd (hunk 1 above)
Comment (HeahDude) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd]: Thank you fixed.

**Review by xabbuh — APPROVED** [review]

**Comment by HeahDude** [comment]
I've also fixed the test.

**Comment by HeahDude** [comment]
I've also fixed the test https://github.com/symfony/symfony/pull/20490/files#diff-aa781e95d1e8284ae37b68430bb2128bR31.

File: src/Symfony/Component/Workflow/Validator/WorkflowValidator.php [hunk 2]
```diff
@@ -20,5 +20,6 @@ class WorkflowValidator implements DefinitionValidatorInterface
 {
     public function validate(Definition $definition, $name)
     {
+        return true;
```
Comment (xabbuh) [inline src/Symfony/Component/Workflow/Validator/WorkflowValidator.php]: I think I tend to agree with @ro0NL that this method shouldn't return anything. Why is that method empty in the `WorkfowValidator` at all?

Comment (HeahDude) [inline src/Symfony/Component/Workflow/Validator/WorkflowValidator.php]: Ok I just remove this extra commit to keep the focus on the config fix in this PR.

Comment (HeahDude) [inline src/Symfony/Component/Workflow/Validator/WorkflowValidator.php]: Done.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 3]
```diff
@@ -420,15 +420,19 @@ private function registerWorkflowConfiguration(array $workflows, ContainerBuilde
...
+            $definitionDefinition = (new Definition(Workflow\Definition::class))
+                ->setPublic(false)
+                ->addArgument($workflow['places'])
+                ->addArgument($transitions)
+                ->addTag('workflow.definition', array(
+                    'name' => $name,
+                    'type' => $type,
+                    'marking_store' => isset($workflow['marking_store']['type']) ? $workflow['marking_store']['type'] : null,
+                ))
+            ;
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: Not sure if it's really worth making this change. It makes `git blame` pretty useless when you try to figure out when a certain change was introduced without much value.

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd [hunk 4]
```diff
@@ -238,6 +238,7 @@
         <xsd:sequence>
             <xsd:element name="marking-store" type="marking_store" />
             <xsd:element name="supports" type="xsd:string" minOccurs="1" maxOccurs="unbounded" />
+            <xsd:element name="initial_place" type="xsd:string" minOccurs="0" maxOccurs="1" />
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd]: should be an attribute, not a child tag

Comment (HeahDude) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd]: Fixed, thanks

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd [hunk 5]
```diff
@@ -235,6 +235,7 @@
     </xsd:complexType>
 
     <xsd:complexType name="workflow">
+        <xsd:attribute name="initial_place" type="xsd:string" />
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd]: I would move it after the sequence next to the already existing attribute.

Comment (HeahDude) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd]: Missed that, thanks!

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php (hunk 3 above)
Comment (lyrixx) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: I agree with @xabbuh

**Review by lyrixx — CHANGES_REQUESTED** [review]

**Comment by lyrixx** [comment]
@HeahDude Could you finish this one? I want to merge it asap.

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd [hunk 6]
```diff
@@ -242,6 +242,7 @@
             <xsd:element name="transitions" type="transitions" />
         </xsd:sequence>
         <xsd:attribute name="name" type="xsd:string" />
+        <xsd:attribute name="initial_place" type="xsd:string" />
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd]: the attribute name should use a dash in XML, not an underscore

**Review by stof — CHANGES_REQUESTED** [review]

**Comment by HeahDude** [comment]
@lyrixx done!

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd (hunk 6 above)
Comment (HeahDude) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/schema/symfony-1.0.xsd]: fixed

**Review by lyrixx — APPROVED** [review]

**Comment by lyrixx** [comment]
:+1:

**Comment by lyrixx** [comment]
Good catch, thanks @HeahDude.

**Pull request merged**